pkg/didcomm/messenger/messenger.go

```diff
@@ -77,16 +81,84 @@ func (m *Messenger) HandleInbound(msg service.DIDCommMsgMap, myDID, theirDID str
 		return fmt.Errorf("threadID: %w", err)
 	}
 
+	if err := m.populateMetadata(thID, msg); err != nil {
+		return fmt.Errorf("with metadata: %w", err)
+	}
+
 	// saves message payload
 	return m.saveRecord(msg.ID(), record{MyDID: myDID, TheirDID: theirDID, ThreadID: thID})
 }
 
+func (m *Messenger) saveMetadata(msg service.DIDCommMsgMap) error {
+	if msg[jsonThread] == nil {
+		return nil
+	}
+
+	thread, ok := msg[jsonThread].(map[string]interface{})
+	if !ok || thread == nil {
+		return nil
+	}
+
+	metadata, ok := thread[jsonMetadata].(map[string]interface{})
```

Comment:
If protocols need access to this metadata then `service.DIDCommMsgMap` should have an accesor for it.

Reply:
Done!